rules/prefer-object-define-properties.js

```diff
@@ -0,0 +1,179 @@
+'use strict';
+const isSameReference = require('./utils/is-same-reference.js');
+const {removeArgument} = require('./fix/index.js');
+
+const MESSAGE_ID = 'prefer-object-define-properties';
+const messages = {
+	[MESSAGE_ID]: 'Prefer `{{replacement}}` over multiple `{{value}}`.',
+};
+
+function getPreviousExpression(node, sourceCode) {
+	const {parent} = node;
+	const visitorKeys
+		= sourceCode.visitorKeys[parent.type] || Object.keys(parent);
+
+	for (const property of visitorKeys) {
+		const value = parent[property];
+		if (Array.isArray(value)) {
+			const index = value.indexOf(node);
+
+			if (index !== -1) {
+				return value[index - 1];
+			}
+		}
+	}
+}
+
+function isObjectDefinePropertyOrObjectDefineProperties(node) {
+	if (node?.type !== 'CallExpression' || node.optional) {
+		return false;
+	}
+
+	const { callee, arguments: callArguments} = node;
+	if (
+		callee.type !== 'MemberExpression'
+		|| callArguments.some(({type}) => type === 'RestElement')
+	) {
+		return false;
+	}
+
+	const {computed, optional, object, property} = callee;
+	if (
+		computed
+		|| optional
+		|| object.type !== 'Identifier'
+		|| object.name !== 'Object'
+		|| property.type !== 'Identifier'
+	) {
+		return false;
+	}
+
+
+	if (property.name === 'defineProperty') {
+		return callArguments.length === 3;
+	}
+
+	if (property.name === 'defineProperties') {
+		return callArguments.length === 2;
+	}
+
+	return false;
+}
+
+function getDescriptorsText(callExpression, sourceCode) {
+	const method = callExpression.callee.property.name;
+
+	if (method === 'defineProperty') {
+		let [, property, descriptor] = callExpression.arguments;
+		let propertyText = sourceCode.getText(property);
+		if (property.type !== 'Literal') {
```

Comment:
This is not right, if string is "a.b", the fix will cause syntax error. The value should be a indentifier.

Reply:
Sorry, mistake, we didn't unquote the string, so the string won't cause problem.